src/pod.ts

```diff
@@ -149,9 +149,9 @@ export abstract class AbstractPod extends base.Resource implements IPodSelector,
    */
   public toSubjectConfiguration(): rb.SubjectConfiguration {
 
-    if (!this.serviceAccount && !this.automountServiceAccountToken) {
```

Comment:
Why did you remove the condition? `automountServiceAccountToken` can still be `true` in case the user configured it explicitly